osu.Game/Screens/Menu/MainMenu.cs

```diff
@@ -117,14 +126,34 @@ protected override bool OnExiting(Screen next)
             return base.OnExiting(next);
         }
 
+        private void promptExit()
+        {
+            if (confirmExit)
+            {
+                dialog?.Push(new ExitConfirmDialog()

```

Comment:
this is wrong. you're doing a null check but in the case it fails what happens? nothing.